index.js

```diff
@@ -15,6 +15,62 @@
 const Bindings = require('bindings')('nodemgclient');
 const pjson = require('./package.json');
 
+// The purpose of create functions is to simplify creation of Memgraph specific
+// data types, e.g. temporal types.
+
+/**
+  * Create Memgraph compatible date object.
+  * @param {number} days - The number of days since 1970-01-01 (unix epoch).
+  */
+function createMgDate(days) {
+  return {
+    "objectType": "date",
+    "days": days,
+  }
+}
+
+/**
+  * Create Memgraph compatible local time object.
+  *
+  * @param {number} nanoseconds - The number of nanoseconds since midnight.
+  */
+function createMgLocalTime(nanoseconds) {
+  return {
+    "objectType": "local_time",
+    "nanoseconds": nanoseconds,
+  }
+}
+
+/**
+  * Create Memgraph compatible local date time object.
+  * @param {number} seconds - The number of seconds since 1970-01-01 (unix epoch).
+  * @param {number} nanoseconds - The number of nanoseconds since the last second.
+  */
+function createMgLocalDateTime(seconds, nanoseconds) {
+  return {
+    "objectType": "local_date_time",
+    "seconds": seconds,
+    "nanoseconds": nanoseconds,
+  }
+}
+
+/**
+  * Create Memgraph compatible duration object.
+  * NOTE: Semantically Memgraph duration is a sum of all
```

Comment:
That's an implementation detail! Maybe remove the comments (we store everything in microseconds) but I don't see how that information is useful to the clients? Am I missing something?

Reply:
I think it's not just an implementation detail; the question is, what's the meaning of these three fields? Now I see it's also wrong here... I'll update + we can discuss :smile:

Reply:
I got a bit confused here... Since we don't have an equivalent concept in the Node.js case, the question is, are these two:
```
pub(crate) fn mg_value_duration(mg_value: *const bindings::mg_value) -> Duration {
    let c_duration = unsafe { bindings::mg_value_duration(mg_value) };
    let days = unsafe { bindings::mg_duration_days(c_duration) };
    let seconds = unsafe { bindings::mg_duration_seconds(c_duration) };
    let nanoseconds = unsafe { bindings::mg_duration_nanoseconds(c_duration) };
    Duration::days(days) + Duration::seconds(seconds) + Duration::nanoseconds(nanoseconds)
}

pub(crate) fn duration_to_mg_duration(input: &Duration) -> *mut bindings::mg_duration {
    let nsec_in_sec = 1_000_000_000_i64;
    let mut nanoseconds = input.num_nanoseconds().unwrap();
    let days = input.num_days();
    nanoseconds -= days * 24 * 60 * 60 * nsec_in_sec;
    let seconds = input.num_seconds() - days * 24 * 60 * 60;
    nanoseconds -= seconds * nsec_in_sec;
    unsafe { bindings::mg_duration_make(0, days, seconds, nanoseconds) }
}
```
(I'm borrowing the Rust code here), valid implementations. If yes, I think a user has to be aware of that, it doesn't matter how it's implemented, but it matters what the `nanoseconds` field is about, or what's the relation between `seconds` and `nanoseconds`.

Reply:
If Duration is rust Duration, that is, https://doc.rust-lang.org/stable/std/time/struct.Duration.html then yes the implementation is correct. There might be a slight mistake, I think `input.num_nanoseconds() should be input.as_nanos()`.  

For the rest, this is a serialization format of packstream and IS8601 which are an implementation detail. Moreover, we slightly diverge from that (we don't fully support ISO 8601 for Duration -- only in a way for the neo4j driver to work with memgraph)

Reply:
I understand that, but it's not about serialization, it's about what to do with these objects, how does a user know what the object is? With Rust is easy because there is documentation of the type (this https://docs.rs/chrono/0.4.19/chrono/struct.Duration.html), but in the Node.js case, we have a custom/new object which nobody understands. That's what I had/have in mind :smile: